lib/cli.js

```diff
@@ -314,9 +316,12 @@ const cli = {
         if (await printResults(engine, resultsToPrint, options.format, options.outputFile)) {
 
             // Errors and warnings from the original unfiltered results should determine the exit code
-            const { errorCount, warningCount } = countErrors(results);
+            const { errorCount, fatalErrorCount, warningCount } = countErrors(results);
+
             const tooManyWarnings =
                 options.maxWarnings >= 0 && warningCount > options.maxWarnings;
+            const fatalErrorExists =
```

Comment:
This variable seems incorrectly named because it doesn’t just indicate that a fatal error exists, it indicates if a fatal error exists AND the option is enabled to cause that to return 2. 

```suggestion
            const shouldExitForFatalErrors =
```